File: compat/src/main/java/eu/darken/porter/compat/PorterIdentity.java
package eu.darken.porter.compat;

import android.content.Context;
import android.content.pm.PackageManager;

final class PorterIdentity {
    static final String PACKAGE = "eu.darken.porter";

    static boolean isInstalled(Context context) {
        return context.getPackageManager().checkSignatures(context.getPackageName(), PACKAGE)
                == PackageManager.SIGNATURE_MATCH;
    }

    private PorterIdentity() {}
}

File: compat/src/main/java/eu/darken/porter/compat/PorterIdentity.kt
package eu.darken.porter.compat

import android.content.Context
import android.content.pm.PackageManager

internal object PorterIdentity {
    const val PACKAGE = "eu.darken.porter"

    fun isInstalled(context: Context): Boolean =
        context.packageManager.checkSignatures(context.packageName, PACKAGE) == PackageManager.SIGNATURE_MATCH
}
